pkg/controller/user-cluster-controller-manager/resources/controller.go

```diff
@@ -190,6 +193,27 @@ func Add(
 		}
 	}
 
+	// Watch cluster if user cluster MLA is enabled so that controller can get resource requirements for user cluster MLA components.
+	if r.userClusterMLA.Monitoring || r.userClusterMLA.Logging {
+		seedWatch := &source.Kind{Type: &kubermaticv1.Cluster{}}
+		if err := seedWatch.InjectCache(seedMgr.GetCache()); err != nil {
+			return fmt.Errorf("failed to inject cache in seed cluster watch for cluster: %w", err)
```

Comment:
Can you leave a comment explaining why you had to inject the cache? I might be dense, but I can't quite explain it off the top of my head.